### PR TITLE
Correct the documentation

### PR DESCRIPTION
**Pull request by bzsolt**
- Fix some style issue, typos, and examples
- Follow the variable naming conventions
- Fix tables both in the project and on the webpage

JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: docs/01.GETTING-STARTED.md [hunk 1]
```diff
@@ -77,72 +80,65 @@ For example the cross-compile to RaspberryPi 2 is something like this:
...
-##### To get a list of all the available buildoptions for Linux:
+**To get a list of all the available buildoptions for Linux**
 
 ```bash
 python tools/build.py --help
 ```
 
-### Checking patch
+
+## Checking patch
```
Comment (zherczeg) [inline docs/01.GETTING-STARTED.md]: Two newlines here?

File: docs/03.API-EXAMPLE.md [hunk 2]
```diff
@@ -214,6 +214,7 @@ print_value (const jerry_value_t value)
     jerry_char_t str_buf_p[req_sz];
 
     jerry_string_to_char_buffer (value, str_buf_p, req_sz);
+    str_buf_p[req_sz] = '\0';
```
Comment (zherczeg) [inline docs/03.API-EXAMPLE.md]: Isn't the jerry_char_t str_buf_p[req_sz]; declaration above is a C++11 feature?

File: docs/03.API-EXAMPLE.md [hunk 3]
```diff
@@ -283,6 +284,12 @@ main (int argc, char * argv[])
       len++;
     }
 
+    /* If the command starts with 'quit', break the loop */
+    if (!strncmp (cmd, "quit", strlen ("quit")))
```
Comment (zherczeg) [inline docs/03.API-EXAMPLE.md]: I think we can use strcmp simply and break only for "quit". Or is there a reason for this prefix check?

File: docs/04.INTERNALS.md [hunk 4]
```diff
@@ -135,57 +135,61 @@ Byte-codes can be placed into four main categories.
 
 Byte-codes of this category serve for placing objects onto the stack. As there are many instructions representing multiple atomic tasks in CBC, there are also many instructions for pushing objects onto the stack according to the number and the type of the arguments. The following table list a few of these opcodes with a brief description.
 
-<div class="CSSTableGenerator" markdown="block">
+<span class="CSSTableGenerator" markdown="block">
 
-| byte-code                            | description                                       |
+| byte-code                            | description |
+| ------------------------------------ | ----------- |
```
Comment (zherczeg) [inline docs/04.INTERNALS.md]: I think the pipes should be aligned.

File: docs/04.INTERNALS.md [hunk 5]
```diff
@@ -135,57 +135,61 @@ Byte-codes can be placed into four main categories.
...
 
 ### Call Byte-codes
 
 The byte-codes of this category perform calls in different ways.
 
-<div class="CSSTableGenerator" markdown="block">
+<span class="CSSTableGenerator" markdown="block">
 
 | byte-code                            | description |
+| ------------------------------------ | ----------- |
```
Comment (zherczeg) [inline docs/04.INTERNALS.md]: Ditto

File: docs/01.GETTING-STARTED.md (hunk 1 above)
Comment (bzsolt) [inline docs/01.GETTING-STARTED.md]: Just a left over, thanks.

File: docs/03.API-EXAMPLE.md (hunk 2 above)
Comment (bzsolt) [inline docs/03.API-EXAMPLE.md]: It's possible, but this declaration method is also used in [main-unix.c:453](https://github.com/Samsung/jerryscript/blob/master/jerry-main/main-unix.c#L453)

File: docs/03.API-EXAMPLE.md (hunk 3 above)
Comment (bzsolt) [inline docs/03.API-EXAMPLE.md]: Yes, because the `cmd` contains a `\n` at the end and the `strcmp` doesn't find them equal, so this is the reason why the `strncmp` is used instead of `strcmp`.

File: docs/04.INTERNALS.md (hunk 4 above)
Comment (bzsolt) [inline docs/04.INTERNALS.md]: Sorry, but I don't fully understand, how should I align the pipes? To the bigger column in the table or to the average line width? Now they are aligned to the average line width.

**Comment by LaszloLango** [comment]
LGTM

File: docs/03.API-EXAMPLE.md (hunk 2 above)
Comment (zherczeg) [inline docs/03.API-EXAMPLE.md]: Fine then.

File: docs/03.API-EXAMPLE.md (hunk 3 above)
Comment (zherczeg) [inline docs/03.API-EXAMPLE.md]: But something like quitten also quits. Perhaps we could check "quit" or "quit\n" and be happy.

Comment (bzsolt) [inline docs/03.API-EXAMPLE.md]: Ok, I've updated the PR according your review.

File: jerry-libc/jerry-libc.c [hunk 6]
```diff
@@ -176,7 +176,7 @@ CALL_PRAGMA (GCC diagnostic pop)
 
 /** Compare two strings. return an integer less than, equal to, or greater than zero
      if s1 is found, respectively, to be less than, to match, or be greater than s2.  */
-int
+int __attr_used___
 strcmp (const char *s1, const char *s2)
```
Comment (bzsolt) [inline jerry-libc/jerry-libc.c]: We should add this function attribute, otherwise the compiler discards `strcmp`.

Comment (zherczeg) [inline jerry-libc/jerry-libc.c]: Is this fix the -O3 issue?

Comment (zherczeg) [inline jerry-libc/jerry-libc.c]: Perhaps we should land this as a different patch, since it is not related to documentation.

Comment (bzsolt) [inline jerry-libc/jerry-libc.c]: This modification is unrelated to the -O3 issue. It's needed for this example, because compile of this code produces the following message: `'strcmp' referenced in section '.text' of <tmp object file> defined in discarded section '.text' of jerry-libc.c.o (symbol from plugin)`
Note: If the libraries and this example compiled without LTO, everything works fine.
I would like to provide a working example with default settings, so I would restore the solution with `strncmp`.
I agree with you, your example (after `> quitten = 3` the program will terminate) shows that this is not the best solution, but this is a simple example code. And force the compiler to keep this symbol at all costs is also not the best solution.
What do you think?

Comment (zherczeg) [inline jerry-libc/jerry-libc.c]: strncmp(str, "quit", strlen("quit") + 1)

All strings are terminated by \0 in C, so adding one to the strlen is ok

Comment (bzsolt) [inline jerry-libc/jerry-libc.c]: I've updated this part.

**Comment by zherczeg** [comment]
LGTM
